docs/changes/io.fits/11922.bugfix.rst

```diff
@@ -0,0 +1 @@
+Implemented ``np.float32`` instance checking for FITS headers values. A ``ValueError`` is now raised if an ``np.float32`` NaN/Inf value is assigned to a header.
```

Comment:
I think the first sentence is not useful and can be confusing since the change only concerns NaN/Inf. Suggestion:
```suggestion
Raise ``ValueError`` if an ``np.float32`` NaN/Inf value is assigned to a header keyword.
```

Reply:
I went back-and-forth on my feelings about that, but I'm inclined to agree.